dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAITextToAudioExecutionSettings.cs

```diff
@@ -0,0 +1,130 @@
+    /// </summary>
+    [JsonPropertyName("voice")]
+    public string Voice
+    {
+        get => this._voice;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._voice = value;
+        }
+    }
+
+    /// <summary>
+    /// The format to audio in. Supported formats are mp3, opus, aac, and flac.
+    /// </summary>
+    [JsonPropertyName("response_format")]
+    public string ResponseFormat
+    {
+        get => this._responseFormat;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._responseFormat = value;
+        }
+    }
+
+    /// <summary>
+    /// The speed of the generated audio. Select a value from 0.25 to 4.0. 1.0 is the default.
+    /// </summary>
+    [JsonPropertyName("speed")]
+    public float Speed
+    {
+        get => this._speed;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._speed = value;
+        }
+    }
+
+    /// <summary>
+    /// Creates an instance of <see cref="AzureOpenAITextToAudioExecutionSettings"/> class with default voice - "alloy".
+    /// </summary>
+    public AzureOpenAITextToAudioExecutionSettings()
+        : this(DefaultVoice)
+    {
+    }
+
+    /// <summary>
+    /// Creates an instance of <see cref="AzureOpenAITextToAudioExecutionSettings"/> class.
+    /// </summary>
+    /// <param name="voice">The voice to use when generating the audio. Supported voices are alloy, echo, fable, onyx, nova, and shimmer.</param>
+    public AzureOpenAITextToAudioExecutionSettings(string voice)
+    {
+        this._voice = voice;
+    }
+
+    /// <inheritdoc/>
+    public override PromptExecutionSettings Clone()
+    {
+        return new AzureOpenAITextToAudioExecutionSettings(this.Voice)
+        {
+            ModelId = this.ModelId,
+            ExtensionData = this.ExtensionData is not null ? new Dictionary<string, object>(this.ExtensionData) : null,
+            Speed = this.Speed,
+            ResponseFormat = this.ResponseFormat
+        };
+    }
+
+    /// <summary>
+    /// Converts <see cref="PromptExecutionSettings"/> to derived <see cref="AzureOpenAITextToAudioExecutionSettings"/> type.
+    /// </summary>
+    /// <param name="executionSettings">Instance of <see cref="PromptExecutionSettings"/>.</param>
+    /// <returns>Instance of <see cref="AzureOpenAITextToAudioExecutionSettings"/>.</returns>
+    public static AzureOpenAITextToAudioExecutionSettings FromExecutionSettings(PromptExecutionSettings? executionSettings)
+    {
+        if (executionSettings is null)
+        {
+            return new AzureOpenAITextToAudioExecutionSettings();
+        }
+
+        if (executionSettings is AzureOpenAITextToAudioExecutionSettings settings)
+        {
+            return settings;
+        }
+
+        var json = JsonSerializer.Serialize(executionSettings);
+
+        var azureOpenAIExecutionSettings = JsonSerializer.Deserialize<AzureOpenAITextToAudioExecutionSettings>(json, JsonOptionsCache.ReadPermissive);
```

Comment:
If someone has existing code that uses `OpenAITextToAudioExecutionSettings` then this serialize/derserialize operation should convert the settings. Is that correct?

Reply:
Yes, correct, and as discussed offline the code should work. Anyway, we'll doble check it in scope of this issue - https://github.com/microsoft/semantic-kernel/issues/7110